File: palindrome.cpp
#include<iostream>
using namespace std;
int main(){
    int num;
    cout<<"enter the number : ";
    cin>>num;

    int n=num;
    int count=0;

    while(num>0){
        int rem=num%10;
        count=count*10+rem;
        num=num/10;
    }
    if(count==n){
        cout<<"the given number is palindrome";
    }
    else{
        cout<<"not a palindrome number";
    }



}

File: palindrome.java
import java.util.*;

public class palindrome {
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the number : ");
        int a=sc.nextInt();
        int comp=a;
        int count=0;

        while (a>0) {
            int rem=a%10;
            count=count*10+rem;
            a=a/10;
        }
        if (count==comp){
            System.out.println("the given number is a palindrome");
        }
        else{
            System.out.println("the number is not a palindrome");
        }

    }
}
